eth/eventdatahandler/handlers.go

```diff
@@ -0,0 +1,550 @@
+package eventdatahandler
+
+import (
+	"bytes"
+	"encoding/hex"
+	"errors"
+	"fmt"
+
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	spectypes "github.com/bloxapp/ssv-spec/types"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/herumi/bls-eth-go-binary/bls"
+	"go.uber.org/zap"
+
+	"github.com/bloxapp/ssv/eth/contract"
+	"github.com/bloxapp/ssv/eth/eventdb"
+	"github.com/bloxapp/ssv/eth1/abiparser"
+	"github.com/bloxapp/ssv/logging/fields"
+	beaconprotocol "github.com/bloxapp/ssv/protocol/v2/blockchain/beacon"
+	ssvtypes "github.com/bloxapp/ssv/protocol/v2/types"
+	registrystorage "github.com/bloxapp/ssv/registry/storage"
+	"github.com/bloxapp/ssv/utils/rsaencryption"
+)
+
+// TODO: review and make sure that we log and return no error in each case where we don't want to stop share processing (e.g. malformed event)
+
+// b64 encrypted key length is 256
+const encryptedKeyLength = 256
+
+func (edh *EventDataHandler) handleOperatorAdded(txn eventdb.RW, event *contract.ContractOperatorAdded) error {
+	logger := edh.logger.With(
+		fields.OperatorID(event.OperatorId),
+		// TODO: move to fields package (check other places in this file)
+		zap.String("operator_pub_key", string(event.PublicKey)),
+		zap.String("owner_address", event.Owner.String()),
+	)
+	// TODO: consider having something like `logger.Info("processing event", zap.String("kind", OperatorAdded))` in the whole package
+	logger.Info("processing OperatorAdded event")
+
+	od := &registrystorage.OperatorData{
+		PublicKey:    event.PublicKey,
+		OwnerAddress: event.Owner,
+		ID:           event.OperatorId,
+	}
+
+	// throw an error if there is an existing operator with the same public key and different operator id
+	if edh.operatorData.ID != 0 && bytes.Equal(edh.operatorData.PublicKey, event.PublicKey) && edh.operatorData.ID != event.OperatorId {
+		edh.logger.Warn("malformed OperatorAdded event: operator registered with the same operator public key")
+		return nil
```

Comment:
shouldn't this be an error then?

Reply:
Handlers are designed in a way that returned error interrupts the flow. So it returns only critical errors like DB